src/Endpoint/AbstractEndpoint.php

```diff
@@ -63,6 +123,21 @@ protected function execute($method, $verb, array $data)
     }
 
     /**
+     * Returns the response for a synchronous request
+     *
+     * @param string $method HTTP method to use for the request
+     * @param string $verb   ThisData verb to be used on this endpoint
+     * @param array  $data   Request body data containing event metadata
+     */
+    protected function synchronousExecute($method, $verb, array $data = [])
```

Comment:
Yeah I wasn't sure how to achieve that with my limited PHP know how. I figured there'd be a better way, but yeah this works. And since it's `protected`, hopefully noone will use it, and we can tidy up the behaviour there later.

To me, it didn't make sense that you'd call `verify` or `getEvents` asynchronously, as the results are used. Though on reflection that's not always true. Happy to keep it as-is until someone complains though :P